libs/payments/ui/src/lib/client/components/CheckoutForm/index.tsx

```diff
@@ -283,7 +283,7 @@ export function CheckoutForm({
                 className="mt-6"
                 createOrder={async () => getPayPalCheckoutToken(cart.currency)}
                 onApprove={async (data: { orderID: string }) => {
-                  await checkoutCartWithPaypal(
+                  checkoutCartWithPaypal(
```

Comment:
This action should still be awaited. This will ensure that the cart.state is set to `processing` before the `redirect` happens.

Reply:
`checkoutCartWithPaypal` can take a while to return, causing the page to seem like its hanging. I'll see if I can rework the action to return quicker, and remove the redirect below this